server/packet_handlers.py

```diff
@@ -188,6 +192,59 @@ async def request_status_update_handler(session: "Session", packet_data: bytes):
     )
 
 
+@bancho_handler(packets.ClientPackets.CHANNEL_PART)
+async def user_leaves_channel_handler(session: "Session", packet_data: bytes):
+    packet_reader = packets.PacketReader(packet_data)
+    channel_name = packet_reader.read_string()
+
+    channel = await channels.fetch_one_by_name(channel_name)
+
+    if not channel:
+        return
+
+    await channel_members.remove(channel["channel_id"], session["session_id"])
+
+
+@bancho_handler(packets.ClientPackets.CHANNEL_JOIN)
+async def user_joins_channel_handler(session: "Session", packet_data: bytes):
+    packet_reader = packets.PacketReader(packet_data)
+    channel_name = packet_reader.read_string()
+
+    channel = await channels.fetch_one_by_name(channel_name)
+
+    if not channel:
+        return
+
+    if session["session_id"] in await channel_members.members(channel["channel_id"]):
+        return
```

Comment:
would like these operations separately for clarity
```suggestion
    channel_members = await channel_members.members(channel["channel_id"])

    if session["session_id"] in channel_members :
        return
```